### DISCUSSION TITLE
about config file & theme

### DISCUSSION THREAD

**Discussion by ovwxxwvo** [fcitx/fcitx5#1206] [Ideas]
Hi, thanks for help me out with `kodi` issue last time.  
嗨，谢谢上次帮我搞定`kodi`输入法的问题。  

I would like to ask why the config file is not placed in the usually used `./config` file folder?  
I am a little puzzled about this and hope to get your explanation.  
Maybe just igore this question. I do a soft link.  
& I just make some input-pane color theme, maybe give some suggestion .  
我想问下小企鹅为啥不把配置文件放在通常目录？  
可否解答一下我的疑惑，也许无视这个问题。我做了软链接。  
给输入栏作了一些颜色主题，可以的话给点建议。  
![misty.fcitx5](https://github.com/ovwxxwvo/misty.fcitx5)
![20241212_134101](https://github.com/user-attachments/assets/d2175fe1-65fc-432f-8e77-014bc76c7f59)

----  

Is that possible to seperate out some color option to one file.  
是否能把颜色选项单独出来到一个文件。  

~/.local/share/fcitx5/themes/misty-dark/theme.conf  
```  
[InputPanel]  
HighlightCandidateColor=#d0ffa0  
NormalColor=#80b050  
PageButtonAlignment=Last Candidate  

[InputPanel/Highlight]  
Color=#00606020  
[InputPanel/Background]  
Color=#002020e0  
BorderColor=#10101080  
BorderWidth=1  
```  

----  

Is that possible to combine tray text color option to theme file.  
是否能把任务栏图标文本颜色选项合并到主题文件。  

~/.local/share/fcitx5/conf/classicui.conf  
```  
Font="JetBrainsMono Nerd Font 14"  
MenuFont="JetBrainsMono Nerd Font 14"  
TrayFont="JetBrainsMono Nerd Font Bold 12"  

TrayTextColor=#d0ffa0  
TrayOutlineColor=#002020  
```  

----

**Comment by wengxt** [comment]
1. Changing font size is a very common use case. If put it within theme, like fcitx4 does, it's too hidden for user to change it.
2. Certain option is at global level, not theme level, is based on the same reason. Usually, tray icon color is more relevant to how your desktop panel themed, not based on fcitx own theme.
3. ~/.config , or ~/.local/share are both common options.
XDG_DATA_DIR is more commonly used for theme data. E.g. /usr/share/icons/ for icon theme /usr/share/themes for gtk & window decoration etc. During fcitx4 era, everything was put under .config, which doesn't really follow the XDG guideline. This was changed in fcitx5 intentionally.

**Reply by ovwxxwvo** [reply]
Thanks, Got it. But ...
